src/code/z_sram_NES.c

```diff
@@ -1,67 +1,1644 @@
+            } else {
+                sramCtx->status = 4;
+            }
+        }
+    } else if (((osGetTime() - sramCtx->unk_18) * 0x40) / 3000 / 10000 >= 200) {
+        sramCtx->status = 0;
+        bzero(*sramCtx->saveBuf, sizeof(*sramCtx->saveBuf));
+        gSaveContext.save.isOwlSave = 0;
+        gSaveContext.save.checksum = 0;
+        // flash read to buffer then copy to save context
+        func_80185968(*sramCtx->saveBuf, sramCtx->curPage, sramCtx->numPages);
+        Lib_MemCpy(&gSaveContext, sramCtx->saveBuf, 0x3CA0);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/Sram_Alloc.s")
+void func_80147314(SramContext* sramCtx, s32 fileNum) {
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/func_80146EBC.s")
+    gSaveContext.save.isOwlSave = false;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/func_80146EE8.s")
+    gSaveContext.save.playerData.newf[0] = '\0';
+    gSaveContext.save.playerData.newf[1] = '\0';
+    gSaveContext.save.playerData.newf[2] = '\0';
+    gSaveContext.save.playerData.newf[3] = '\0';
+    gSaveContext.save.playerData.newf[4] = '\0';
+    gSaveContext.save.playerData.newf[5] = '\0';
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/func_80146F5C.s")
+    gSaveContext.save.checksum = 0;
+    gSaveContext.save.checksum = Sram_CalcChecksum(&gSaveContext, 0x3CA0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/func_80147008.s")
+    Lib_MemCpy(*sramCtx->saveBuf, &gSaveContext, 0x3CA0);
+    func_80146EBC(sramCtx, D_801C6840[fileNum * 2], D_801C6850[fileNum * 2]);
+    func_80146EBC(sramCtx, D_801C6840[fileNum * 2 + 1], D_801C6850[fileNum * 2]);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/func_80147020.s")
+    gSaveContext.save.isOwlSave = true;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/func_80147068.s")
+    gSaveContext.save.playerData.newf[0] = 'Z';
+    gSaveContext.save.playerData.newf[1] = 'E';
+    gSaveContext.save.playerData.newf[2] = 'L';
+    gSaveContext.save.playerData.newf[3] = 'D';
+    gSaveContext.save.playerData.newf[4] = 'A';
+    gSaveContext.save.playerData.newf[5] = '3';
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/func_80147138.s")
+void func_80147414(SramContext* sramCtx, s32 fileNum, s32 arg2) {
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/func_80147150.s")
+    // Clear save buffer
+    bzero(*sramCtx->saveBuf, sizeof(*sramCtx->saveBuf));
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/func_80147198.s")
+    // Read save file
+    if (func_80185968(*sramCtx->saveBuf, D_801C6840[fileNum * 2], D_801C6850[fileNum * 2]) != 0) {
+        // If failed, read backup save file
+        func_80185968(*sramCtx->saveBuf, D_801C6840[fileNum * 2 + 1], D_801C6850[fileNum * 2 + 1]);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/func_80147314.s")
+    // Copy buffer to save context
+    Lib_MemCpy(&gSaveContext, sramCtx->saveBuf, 0x3CA0);
```

Comment:
What size is `0x3CA0` suppose to represent? Any thoughts? Is `u8 (*saveBuf)[0x4000];` suppose to be `u8 (*saveBuf)[0x3CA0];`?

Reply:
It is basically copying all this chunk of the SaveContext struct:
```c
    /* 0x0000 */ Save save;
    /* 0x100C */ u8 eventInf[8];
    /* 0x1014 */ u8 unk_1014;                    // original name: stone_set_flag
    /* 0x1015 */ u8 unk_1015;
    /* 0x1016 */ u16 jinxTimer;
    /* 0x1018 */ s16 rupeeAccumulator;           // original name: lupy_udct
    /* 0x101A */ u8 unk_101A[6];                 // original name: bottle_status, one entry for each bottle
    /* 0x1020 */ OSTime unk_1020[6];             // original name: bottle_ostime, one entry for each bottle
    /* 0x1050 */ OSTime unk_1050[6];             // original name: bottle_sub, one entry for each bottle
    /* 0x1080 */ OSTime unk_1080[6];             // original name: bottle_time, one entry for each bottle
    /* 0x10B0 */ OSTime unk_10B0[6];             // original name: bottle_stop_time, one entry for each bottle
    /* 0x10E0 */ u64 pictoPhoto[1400];           // buffer containing the pictograph photo
```

This implies that chunk of the SaveContext was yet another substruct (and that number probably comes from `sizeof(thatSubstruct)`). I decided to not make that substruct  because it would bloat too much the accesses to common stuff of gSaveContext